dialog/doc.go

```diff
@@ -0,0 +1,5 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.md file in the project root for full license information.
+
+// Package dialog provides functionality for client-service speech and activity interactions
```

Comment:
Shouldn't we talk about the whole virtual assistant stuff here?

Reply:
How about this?

// Package dialog provides functionality for creating custom voice assistant applications and managing the related interaction flow